### PR TITLE
fix: [EXT-3586] add default cache

### PR DESCRIPTION
**Pull request by loweisz**

### CODE REVIEW

File: src/keys/get-management-token.ts [hunk 1]
```diff
@@ -16,6 +16,8 @@ export interface GetManagementTokenOptions {
   reuseToken?: boolean
 }
 
+const defaultCache = new NodeCache()
```
Comment (shikaan) [inline src/keys/get-management-token.ts:19]: What about 

```
let defaultCache: NodeCache;

export const getManagementToken = (privateKey: string, opts: GetManagementTokenOptions) => {
  if (opts.reuseToken) {
    defaultCache = new NodeCache();
  }
  ...
```

So we don't over-allocate if it's not needed. We could also cleanup `createGetManagementToken` at that point

File: src/keys/get-management-token.ts [hunk 2]
```diff
@@ -142,8 +139,12 @@ export const createGetManagementToken = (
  * @category Keys
  */
 export const getManagementToken = (privateKey: string, opts: GetManagementTokenOptions) => {
-  return createGetManagementToken(createLogger({ filename: __filename }), createHttpClient())(
-    privateKey,
-    opts
-  )
+  if (opts.reuseToken || opts.reuseToken === undefined) {
+    defaultCache = new NodeCache()
```
Comment (Jwhiles) [inline src/keys/get-management-token.ts:143]: I am a bit confused by this. Doesn't it mean that we are initialising the cache each time you a user tries gets a token?

**Review by Jwhiles — CHANGES_REQUESTED** [review]

File: src/keys/get-management-token.ts (hunk 2 above)
Comment (loweisz) [reply src/keys/get-management-token.ts:143]: Yes you are right, I added a check that this only happens once

File: src/keys/get-management-token.ts [hunk 3]
```diff
@@ -142,8 +139,12 @@ export const createGetManagementToken = (
...
  */
 export const getManagementToken = (privateKey: string, opts: GetManagementTokenOptions) => {
-  return createGetManagementToken(createLogger({ filename: __filename }), createHttpClient())(
-    privateKey,
-    opts
-  )
+  if ((opts.reuseToken || opts.reuseToken === undefined) && !defaultCache) {
+    defaultCache = new NodeCache()
+  }
+  return createGetManagementToken(
```
Comment (Jwhiles) [inline src/keys/get-management-token.ts:145]: I guess we should now ask why this function even exists. We could just inline the `createGetManagementToken` into this function, and it probably makes the code slightly easier to follow. I don't think we need to do it in this PR though.

**Review by Jwhiles — APPROVED** [review]

**Review by shikaan — APPROVED** [review]

**Pull request merged**